portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -51,4 +52,11 @@ public String getCommentContent() {
   public Date getDate() {
     return date;
   }
+
+  /**
+   * Translates the comment content in the language specified.
+   */
+  public void translate(String languageCode) {
```

Comment:
I recommend not putting any logic into the entity classes and use them only as a database/network interface. Essentially you have this doing multiple unrelated things now:
 - interfacing with the database and network (which I think is fine at the moment)
 - translating it's own comment

This is also dangerous because you're changing the state of the entity that is directly related to what you read from the database. This can easily lead to weird behavior where sometimes in your program the comment is translated and sometimes it isn't.

Another pattern you could use is creating a `Translator` that takes a commment instance and returns a new instance with the translated comment content (basically making a comment immutable). Alternatively you can change the comment field in there directly on the instance if you think immutable objects are overkill. You can read up more on this by searching for Data Transfer Object (e.g. https://martinfowler.com/eaaCatalog/dataTransferObject.html) and immutability. The views of people go in both directions, as always...

Reply:
I thought that create a new list and new instances for every comment wouldn't be very efficient. Also I wouldn't want to leave the possibility to edit the comment unconditionally, that's why I chose a middle way, which is to be able to edit the comment just to translate it.

Reply:
I agree on the efficiency part but when you put this in relation to how fast/slow database access or network communication is, creating new objects is practically free. You can totally decide to not make your objects immutable but even then it would probably be more appropriate to have some other class handle the actual translation rather than the comment knowing how to translate itself.

Reply:
Can I add the translation of a comment in `TranslationProvider` or is it better to create another class?

Reply:
I think for now they could go into the same class. Especially once you start translating other things than just comments, it starts making more sense to have a separate class and the translator class would only get a string + language and return a string.